lib/linter/linter.js

```diff
@@ -305,7 +305,11 @@ function createDisableDirectives(options) {
 
         // push to directives, if the rule is defined(including null, e.g. /*eslint enable*/)
         if (ruleId === null || !!ruleMapper(ruleId)) {
-            result.directives.push({ parentComment, type, line: commentToken.loc.start.line, column: commentToken.loc.start.column + 1, ruleId });
+            if (type === "disable-next-line") {
+                result.directives.push({ parentComment, type, line: commentToken.loc.end.line, column: commentToken.loc.end.column + 1, ruleId });
```

Comment:
A problem with changing these `line` and `column` is that they're also used as the position to report the directive as unused, so it would look like this:

![image](https://user-images.githubusercontent.com/44349756/147226071-84719ddb-b7e1-483a-9901-acfad3138ead.png)

We could modify the code that creates unused disable directive errors to use `loc` from `parentComment.commentToken` instead of these `line` and `column`, or maybe even consider entirely removing these `line` and `column` and using only `parentComment.commentToken`  in `apply-disable-directives.js` for everything, since the logic of where disable directives apply is in `apply-disable-directives.js` anyway.

Reply:
fixed.